### DISCUSSION TITLE
Do PageSpeed Insight analyze the whole JS or just the code for home page?

### DISCUSSION THREAD

**Discussion by stressed-pet-owner** [gatsbyjs/gatsby#29443] [Help]
I'm having a hard time making my mobile score a consistent 85+. I upgraded to the new gatsby-plugin-image and nothing changes. I went and totally removed my hero image, but still can't get my site to reach the said score. I'm very pleased with my desktop score though. A consistent 95+!

What I noticed is these Apollo codes. I'm pretty sure I only wrapped the component that is not in the home page with the Apollo provider so I don't know why is it there..

![Image 2021-02-11 at 8 21 22 AM](https://user-images.githubusercontent.com/78886814/107591059-c2666b00-6c44-11eb-955b-e650be1c4ddb.png)

**Comment by ascorbic** [comment]
Hi. Pagespeed Insights will only analyze the URL that you pass to it, but `app-xxxxx` is the main bundle which is included on every page. Are you importing it in any component that is also used on the homepage, such as any wrapper component? You could try [webpack-bundle-analyser](https://www.gatsbyjs.com/plugins/gatsby-plugin-webpack-bundle-analyser-v2/) to see what is being included in each component.

Are you able to share a link to the site, because it may be somehting else that's causing the issue.

**Reply by stressed-pet-owner** [reply]
I'm only using it on the Projects page.. link: https://eloquent-bassi-0b077d.netlify.app/

**Reply by stressed-pet-owner** [reply]
Also, I'm having the new gatsby-plugin-image as the LCP. Although I made it display:none on mobile the other time, I still got the same results..

**Reply by ascorbic** [reply]
Hey. So I've looked at your site [in WebPageTest](https://www.webpagetest.org/video/compare.php?tests=210211_DiSF_60ec907dc3cd02ced10a09f7067455d1-r:1-c:0) and have a few suggestions. First, if you look at the filmstrip, you can see that the traced svg isn't a great placeholder at mobile size. I'd suggest changing it to either dominant color, or just setting it to none and setting a background color yourself manually.
Second, the way you're handling the main heading text will be causing problems. Animating it in means that it's not visible for quite a long time, which will cause you problems. You also don't provide a fallback font. I would suggest removing the animation, and setting the font stack to include fallback fonts, with `font-display: swap`.

**Reply by stressed-pet-owner** [reply]
I see... Thank you for this!! But how about the Apollo js files seen in home page? I only used it in the Projects page...

**Reply by stressed-pet-owner** [reply]
Inner pages got good scores on mobile.. problem is just the home page..

**Reply by stressed-pet-owner** [reply]
Hi @ascorbic. Any idea on how can I solve the unused JS and LCP on home page ?

**Reply by ascorbic** [reply]
I still very much recommend removing the text animation from the front page. LCP is your biggest issue on that page, and that's entirely down to that animation.

**Reply by ascorbic** [reply]
Look at the LCP here: https://www.webpagetest.org/video/compare.php?tests=210215_DiAQ_3f4c009819dbaf23e3d90fa560767889-r%3A1-c%3A0&highlightLCP=1&thumbSize=200&ival=500&end=visual

**Reply by stressed-pet-owner** [reply]
@ascorbic I removed the animation on hero and score went to 80. Do you recommend removing animation on the whole page as well? I'm okay with my site having no animation at all, but some clients really want some fancy movements around their site. lol

**Reply by ascorbic** [reply]
Animations that happen on load can be really bad for Lighthouse scores. Others are fine.
